### ISSUE TITLE
POSIX like

### ISSUE THREAD

**Issue by EduardShaid** [jimtremblay/nOS#9]
it would be good to do on top of your RTOS shell POSIX

for example, to make pthread_mutex_lock () as the:
pthread_create (...) {
...
     nOS_ThreadCreate(...);
...
};

pthread_mutex_lock (...) {
...
     nOS_MutexLock(...);
...
};

**Comment by jimtremblay** [comment]
Yeah, a compatible layer can be done on top of nOS to have the possibility to compile your app for a microcontroller or a stronger Linux capable microprocessor.

Maybe you can help if you want.
